tests/phpunit/tests/rest-api/wpRestTemplatesController.php

```diff
@@ -143,11 +146,76 @@ public function test_get_item() {
 				'status'         => 'publish',
 				'wp_id'          => self::$post->ID,
 				'has_theme_file' => false,
+				'is_custom'      => true,
+				'author'         => 0,
 			),
 			$data
 		);
 	}
 
+	/**
+	 * @ticket 54507
+	 * @dataProvider get_template_endpoint_urls
+	 */
+	public function test_get_item_works_with_a_single_slash( $endpoint_url ) {
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'GET', $endpoint_url );
+		$response = rest_get_server()->dispatch( $request );
+
+		$data = $response->get_data();
+		unset( $data['content'] );
+		unset( $data['_links'] );
+
+		$this->assertEquals(
+			array(
+				'id'             => 'tt1-blocks//index',
+				'theme'          => 'tt1-blocks',
+				'slug'           => 'index',
+				'title'          => array(
+					'raw'      => 'Index',
+					'rendered' => 'Index',
+				),
+				'description'    => 'The default template used when no other template is available. This is a required template in WordPress.',
+				'status'         => 'publish',
+				'source'         => 'theme',
+				'type'           => 'wp_template',
+				'wp_id'          => null,
+				'has_theme_file' => true,
+				'is_custom'      => false,
+			),
+			$data
+		);
+	}
+
+	public function get_template_endpoint_urls() {
+		return array(
+			array( '/wp/v2/templates/tt1-blocks/index' ),
+			array( '/wp/v2/templates/tt1-blocks//index' ),
+		);
+	}
```

Comment:
This test uses different template names from the Gutenberg version. Ideally we'd bet both in sync with each other. The following change will get this PR across the line:

```suggestion	/**
	 * @ticket 54507
	 * @dataProvider get_template_endpoint_urls
	 */
	public function test_get_item_works_with_a_single_slash( $endpoint_url ) {
		wp_set_current_user( self::$admin_id );
		$request  = new WP_REST_Request( 'GET', $endpoint_url );
		$response = rest_get_server()->dispatch( $request );

		$data = $response->get_data();
		unset( $data['content'] );
		unset( $data['_links'] );

		$this->assertSame(
			array(
				'id'             => 'default//my_template',
				'theme'          => 'default',
				'slug'           => 'my_template',
				'source'         => 'custom',
				'origin'         => null,
				'type'           => 'wp_template',
				'description'    => 'Description of my template.',
				'title'          => array(
					'raw'      => 'My Template',
					'rendered' => 'My Template',
				),
				'status'         => 'publish',
				'wp_id'          => self::$post->ID,
				'has_theme_file' => false,
				'is_custom'      => true,
				'author'         => 0,
			),
			$data
		);
	}

	public function get_template_endpoint_urls() {
		return array(
			array( '/wp/v2/templates/default/my_template' ),
			array( '/wp/v2/templates/default//my_template' ),
		);
	} ```